daemon/boardd/server.cpp

```diff
@@ -165,14 +165,15 @@ void start_server(const char *host, unsigned short port) {
     std::cerr << "boardd: built with mulit-threading, "
 	<< NUM_THREADS << " threads." << std::endl;
 
-    std::list<std::thread> threads;
     asio::io_service io_service;
     Server server(io_service, host, port);
-    for (int i = 0; i < NUM_THREADS; ++i) {
-	threads.push_back(std::move(
-		std::thread(std::bind(&ServiceThread, std::ref(io_service)))));
+    std::vector<std::future> futures;
+    futures.reserve(NUM_THREADS);
+    for (std::size_t i = 0; i < NUM_THREADS; ++i) {
+	futures.emplace_back(
+		std::async(std::launch::async,ServiceThread,std::ref(io_service)));
```

Comment:
Please preserve the style. Keep it under 80 chars/line, and one space after comma.